packages/opencensus-nodejs/test/test-tracing.ts

```diff
@@ -0,0 +1,203 @@
+  describe('new Tracing()', () => {
+    it('should create a Tracer instance', () => {
+      const tracing = new Tracing();
+      assert.ok(tracing instanceof Tracing);
+    });
+  });
+
+  /** Should get the singleton trancing instance. */
+  describe('static get instance()', () => {
+    it('should get the singleton trancing instance', () => {
+      const tracing = Tracing.instance;
+      assert.ok(tracing instanceof Tracing);
+    });
+  });
+
+  /** Should return a started tracing instance */
+  describe('start()', () => {
+    let aTracing: types.Tracing;
+    const tracing = new Tracing();
+    // tslint:disable:no-any
+    beforeEach(() => {
+      if (tracing.active) {
+        tracing.stop();
+      }
+    });
+
+    it('should return a tracing instance', () => {
+      aTracing = tracing.start();
+      assert.ok(aTracing instanceof Tracing);
+      assert.ok(tracing.active);
+      assert.ok(aTracing.tracer.active);
+    });
+
+    it('should start with default config', () => {
+      tracing.start();
+      assert.strictEqual(defaultConfig.bufferSize, tracing.config.bufferSize);
+      assert.strictEqual(
+          defaultConfig.bufferTimeout, tracing.config.bufferTimeout);
+      assert.strictEqual(defaultConfig.logLevel, tracing.config.logLevel);
+      assert.strictEqual(
+          defaultConfig.maximumLabelValueSize,
+          tracing.config.maximumLabelValueSize);
+      assert.strictEqual(
+          defaultConfig.samplingRate, tracing.config.samplingRate);
+      assert.ok(tracing.config.plugins['http']);
+      assert.strictEqual(
+          tracing.config.plugins['http'],
+          `${Constants.OPENCENSUS_SCOPE}/${
+              Constants.DEFAULT_PLUGIN_PACKAGE_NAME_PREFIX}-http`);
+
+      assert.ok(tracing.config.plugins['https']);
+      assert.strictEqual(
+          tracing.config.plugins['https'],
+          `${Constants.OPENCENSUS_SCOPE}/${
+              Constants.DEFAULT_PLUGIN_PACKAGE_NAME_PREFIX}-https`);
+    });
+
+    it('should start tracing with a non-default logLevel', () => {
+      aTracing = tracing.start({logLevel: 3});
+      assert.strictEqual(tracing.config.logLevel, 3);
+      const consoleLogger = aTracing.tracer.logger as logger.ConsoleLogger;
+      assert.strictEqual(consoleLogger.level, 'info');
+    });
+
+    it('should start tracing with a logger instance', () => {
+      const aLogger = logger.logger('debug');
+      aTracing = tracing.start({logger: aLogger});
+      assert.strictEqual(tracing.config.logger, aLogger);
+      const consoleLogger = aTracing.tracer.logger as logger.ConsoleLogger;
+      assert.strictEqual(consoleLogger.level, 'debug');
+    });
+
+    it('should start with an exporter instance', () => {
+      aTracing = tracing.start({exporter: NOOP_EXPORTER});
+      assert.strictEqual(tracing.config.exporter, NOOP_EXPORTER);
+      assert.strictEqual(aTracing.exporter, NOOP_EXPORTER);
+    });
+
+    it('should start with a non-default bufferSize', () => {
```

Comment:
For all the config object tests, I would suggest grouping them into its own `describe` for configuration initialization (nested within the current one).

Reply:
Done !